### PR TITLE
Inactive votes election hint

### PR DESCRIPTION
**Pull request by guilhermelawless**
Allows nodes to lazily insert an election that is likely to confirm.

### CODE REVIEW

File: nano/core_test/active_transactions.cpp [hunk 1]
```diff
@@ -646,6 +646,164 @@ TEST (active_transactions, inactive_votes_cache_multiple_votes)
 	ASSERT_EQ (2, node.stats.count (nano::stat::type::election, nano::stat::detail::vote_cached));
 }
 
+TEST (active_transactions, inactive_votes_cache_election_start)
+{
+	nano::system system (1);
```
Comment (wezrule) [inline nano/core_test/active_transactions.cpp]: Doesn't seem like you need the node created by the system constructor

Comment (guilhermelawless) [reply nano/core_test/active_transactions.cpp]: Thanks, removed it

**Review by SergiySW — DISMISSED** [review]

File: nano/core_test/active_transactions.cpp [hunk 2]
```diff
@@ -646,6 +646,164 @@ TEST (active_transactions, inactive_votes_cache_multiple_votes)
...
+	// A late block arrival also checks the inactive votes cache
+	ASSERT_TRUE (node.active.empty ());
+	auto send7_cache (node.active.find_inactive_votes_cache (send7->hash ()));
+	ASSERT_EQ (6, send7_cache.voters.size ());
+	ASSERT_TRUE (send7_cache.status.bootstrap_started);
+	ASSERT_TRUE (send7_cache.status.confirmed);
+	ASSERT_TRUE (send7_cache.status.election_started); // already marked even though the block does not exist
+	node.process_active (send6);
+	node.block_processor.flush ();
+	// An election is started for send6 but does not confirm
```
Comment (clemahieu) [inline nano/core_test/active_transactions.cpp:799]: We can add a check here for cemented count to prove send6 did not confirm.

File: nano/core_test/toml.cpp [hunk 3]
```diff
@@ -404,6 +405,7 @@ TEST (toml, daemon_config_deserialize_no_defaults)
 	network_threads = 999
 	online_weight_minimum = "999"
 	online_weight_quorum = 99
+	election_hint_weight_percent = 19
```
Comment (clemahieu) [inline nano/core_test/toml.cpp:408]: This is 19 here, and 10 below. Is this supposed to be 10?

File: nano/node/nodeconfig.hpp [hunk 4]
```diff
@@ -56,6 +56,7 @@ class node_config
 	unsigned vote_generator_threshold{ 3 };
 	nano::amount online_weight_minimum{ 60000 * nano::Gxrb_ratio };
 	unsigned online_weight_quorum{ 50 };
+	unsigned election_hint_weight_percent{ 10 };
```
Comment (clemahieu) [inline nano/node/nodeconfig.hpp:59]: And 10 here. Are these supposed to be using a common constant in case it needs to be changed and stay in sync in the future?

File: nano/node/gap_cache.cpp [hunk 5]
```diff
@@ -88,29 +88,34 @@ bool nano::gap_cache::bootstrap_check (std::vector<nano::account> const & voters
...
-				{
-					node_l->bootstrap_initiator.bootstrap_lazy (hash_a);
-				}
-				else if (!node_l->flags.disable_legacy_bootstrap)
-				{
-					node_l->bootstrap_initiator.bootstrap ();
-				}
-			}
-		});
+		bootstrap_start (hash_a);
```
Comment (clemahieu) [inline nano/node/gap_cache.cpp]: Are there any changes here or just extracting a function? Can this be done in a separate PR to simplify this?

File: nano/node/blockprocessor.cpp [hunk 6]
```diff
@@ -307,6 +307,10 @@ void nano::block_processor::process_live (nano::block_hash const & hash_a, std::
 			election.election->try_generate_votes (block_a->hash ());
 		}
 	}
+	else
+	{
+		node.active.check_inactive_votes_cache_election (block_a);
```
Comment (clemahieu) [inline nano/node/blockprocessor.cpp]: I think the work "trigger" makes more sense than "check" because this is taking an action based on status.

File: nano/node/gap_cache.cpp (hunk 5 above)
Comment (SergiySW) [reply nano/node/gap_cache.cpp]: Just extraction. Can be separate

File: nano/core_test/toml.cpp (hunk 3 above)
Comment (guilhermelawless) [reply nano/core_test/toml.cpp:408]: Oh this is a config test to ensure the values are all correctly deserialized, see the values above which are also not default.

File: nano/node/nodeconfig.hpp (hunk 4 above)
Comment (guilhermelawless) [reply nano/node/nodeconfig.hpp:59]: 10 is the default and can be configured, no constants necessary I think.

File: nano/node/gap_cache.cpp (hunk 5 above)
Comment (guilhermelawless) [reply nano/node/gap_cache.cpp]: Reverted 👍

**Review by clemahieu — APPROVED** [review]

**Review by SergiySW — APPROVED** [review]
